File: Algorithms/src/3689. Maximum Total Subarray Value I/Code/solution.cpp
class Solution
{
public:
    long long maxTotalValue(vector<int> &nums, int k)
    {
        // Find the smallest element in the array
        long long mn = *min_element(nums.begin(), nums.end());

        // Find the largest element in the array
        long long mx = *max_element(nums.begin(), nums.end());

        // Best subarray value = global maximum - global minimum
        long long best = mx - mn;

        // We can choose the same best subarray k times
        return best * k;
    }
};

File: Algorithms/src/3689. Maximum Total Subarray Value I/Code/solution.go
func maxTotalValue(nums []int, k int) int64 {
    // Initialize minimum and maximum
    mn := nums[0]
    mx := nums[0]

    // Find global minimum and maximum
    for _, num := range nums {
        if num < mn {
            mn = num
        }
        if num > mx {
            mx = num
        }
    }

    // Best subarray value
    best := int64(mx - mn)

    // Choose the same best subarray k times
    return best * int64(k)
}